# Perda Penyiaran

Abstrak

- Multilevel Broadcasting:  Policy Blindspots

@prabowoEraPenyiaranDigital2012


In [1]:
import requests
from bs4 import BeautifulSoup as soup
import pandas as pd
from tqdm import tqdm

In [2]:
s = requests.Session()
headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36'
}

In [47]:
# url = 'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor=&jenis=19&jenis=20&jenis=23'
url = 'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor='
r = s.get(url, headers=headers)
res = soup(r.text, 'html.parser')
# print(res)

In [40]:
print(int(res.find('div', attrs={'class':'my-lg-16 my-10'}).find('p').text.split(' ')[1].replace('.', ''))) # jumlah peraturan

2921


In [48]:
last_pages = int(res.find('ul', attrs={'class':'pagination'}).find_all('a', attrs={'class':'page-link'})[-1]['href'].split('=')[-1]) # jumlah halaman
pages = last_pages + 1
print(pages)

294


In [3]:
result = []
data_regulasi = []
meta_regulasi = []

# url = 'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor=&jenis=19&jenis=20&jenis=23'
url = 'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor='
r1 = s.get(url, headers=headers)
res1 = soup(r1.text, 'html.parser')

last_pages = int(res1.find('ul', attrs={'class':'pagination'}).find_all('a', attrs={'class':'page-link'})[-1]['href'].split('=')[-1]) # jumlah halaman
pages = last_pages + 1

for page_number in tqdm(range(1, pages)):
  # url_ = f'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor=&jenis=19&jenis=20&jenis=23&p={page_number}'
  url_ = f'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor=&p={page_number}'
  r2 = s.get(url_, headers=headers)
  res2 = soup(r2.text, 'html.parser')

  for item in res2.find('div', attrs={'class':'rounded-4'}).find_all('div', attrs={'class':'row mb-8'}):

    jenis_no = item.find('div', attrs={'class':'col-lg-8 fw-bold fs-4'}).text.strip() # jenis dan nomor peraturan
    href_detil = item.find('a', attrs={'class':'fs-1 fw-bold'})['href'] # link page_details https://peraturan.bpk.go.id/{page_details}
    judul = item.find('a', attrs={'class':'fs-1 fw-bold'}).text.strip().replace('\r\n', ' ') # Nama /Judul Peraturan
    subjek = '|'.join([x.text for x in item.find_all('span', attrs={'class':'badge'})]) # subjek peraturan
    cuplikan = '|'.join([a.text for a in item.find_all('div', attrs={'class':'mb-8'})]) # cuplikan text
    href_download = item.find('a', attrs={'class':'download-file'})['href'] # href download
    data_id = item.find('a', attrs={'class':'download-file'})['data-id'] # data-id
    # print(item.find('a', attrs={'class':'download-file'}).text.strip()) # nama file
    regulasi = {
        'data_id': data_id,
        'jenis_no': jenis_no,
        'judul': judul,
        'subjek': subjek,
        'cuplikan': cuplikan,
        'href_detil': href_detil,
        'href_unduh': href_download
    }
    data_regulasi.append(regulasi)

  for meta in res2.find_all('div', attrs={'class':'modal-body'}):
    if meta.find('div', attrs={'class':'fs-5'}) != None:
      judul_singkat = meta.find('div', attrs={'class':'fs-5'}).find_all('p')[0].text.strip() # judul_singkat
      tahun = meta.find('div', attrs={'class':'fs-5'}).find_all('p')[1].text.strip() # tahun
      sumber = meta.find('div', attrs={'class':'fs-5'}).find_all('p')[2].text.strip() # sumber
      judul_panjang = meta.find('div', attrs={'class':'fs-5'}).find_all('p')[3].text.strip().replace('\r\n',' ') # judul_panjang

      # Konsideran. mitigate nonetype data
      konsiderans = meta.find_all('tr')[0]
      pertimbangan = konsiderans.find_all('li')[0].text.strip() if len(konsiderans.find_all('li')) == 3 else '' # pertimbangan
      konsideran = konsiderans.find_all('li')[1].text.strip() if len(konsiderans.find_all('li')) == 3 else '' # konsideran
      materi_pokok = konsiderans.find_all('li')[2].text.strip() if len(konsiderans.find_all('li')) == 3 else '' # materi pokok

      # Catatan
      catatan = '\n'.join([x.text for x in meta.find_all('ul', attrs={'class':'dash'})[1].find_all('li')]) # catatan

      metadata = {
          'judul_panjang': judul_panjang,
          'tahun': tahun,
          'sumber': sumber,
          'judul_singkat': judul_singkat,
          'pertimbangan': pertimbangan,
          'konsideran': konsideran,
          'materi_pokok': materi_pokok,
          'catatan': catatan
      }
      meta_regulasi.append(metadata)

# result.append({
#     'data': data_regulasi,
#     'meta': meta_regulasi
# })


100%|██████████| 395/395 [03:18<00:00,  1.99it/s]


In [ ]:
data_regulasi

In [8]:
regulasi_df = pd.DataFrame(data_regulasi)
regulasi_df

data_id                                           jenis_no  \
0      38966        Peraturan Pemerintah (PP) No. 52 Tahun 2005   
1      38952        Peraturan Pemerintah (PP) No. 50 Tahun 2005   
2      38955        Peraturan Pemerintah (PP) No. 51 Tahun 2005   
3      38618        Peraturan Pemerintah (PP) No. 11 Tahun 2005   
4     254819  Peraturan Menteri Komunikasi dan Informatika N...   
...      ...                                                ...   
3940  228833  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
3941  228813  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
3942  309224  Peraturan Gubernur (PERGUB) Provinsi Jawa Timu...   
3943  158222  Peraturan Bupati (PERBUP) Kabupaten Katingan N...   
3944  228834  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   

                                                  judul  \
0     Penyelenggaraan Penyiaran Lembaga Penyiaran Be...   
1     Penyelenggaraan Penyiaran Lembaga Penyiaran Sw...   
2     Penyelenggaraan Penyiaran Lembaga Penyiaran Ko...   
3     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...   
4     Penyelenggaraan Penyiaran Melalui Sistem Stasi...   
...                                                 ...   
3940  Perubahan Kedelapan atas Peraturan Bupati Siak...   
3941  Perubahan Ketiga atas Peraturan Bupati Siak No...   
3942  Perubahan atas Peraturan Gubernur Nomor 32 Tah...   
3943  Penggolongan dan kodifikasi Barang Milik Daera...   
3944  Perubahan Kesembilan atas Peraturan Bupati Sia...   

                                                 subjek  \
0     Informasi, Pers, Pos, dan Periklanan|Diubah de...   
1     Informasi, Pers, Pos, dan Periklanan|Diubah de...   
2     Informasi, Pers, Pos, dan Periklanan|Diubah de...   
3     Informasi, Pers, Pos, dan Periklanan|Diubah de...   
4     Telekomunikasi, Informatika, dan Internet|Dica...   
...                                                 ...   
3940  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   
3941  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   
3942                               APBD|Standar/Pedoman   
3943             Pengelolaan Barang Milik Negara/Daerah   
3944  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   

                                               cuplikan  \
0     ... Peraturan Pemerintah tentang Penyelenggara...   
1     ... Peraturan Pemerintah ini telah memiliki Le...   
2     ... BAB V   RENCANA DASAR TEKNIK DAN PERSYARAT...   
3     ... Siaran, penyiaran, penyiaran radio, penyia...   
4     ... 1 Menimbang : a. bahwa penyiaran diselengg...   
...                                                 ...   
3940  ... Bus Air 613.152 000.00 613.152.000.00 O.00...   
3941  ... Bus Air 613.152,000.00 613.152.000.00 6.00...   
3942  ... Kegiatan 50.000.000       2231 8.1.02.02.0...   
3943  ... STATISTIK 62 6 2-16 0-0 0-0 02 SEKRETARIAT...   
3944  ... Programi 7.575.392.000.00 7.575.302.000,00...   

                                             href_detil  \
0                    /Details/49688/pp-no-52-tahun-2005   
1                    /Details/49674/pp-no-50-tahun-2005   
2                    /Details/49678/pp-no-51-tahun-2005   
3                    /Details/49339/pp-no-11-tahun-2005   
4     /Details/159041/permenkominfo-no-43permkominfo...   
...                                                 ...   
3940   /Details/170256/perbup-kab-siak-no-26-tahun-2020   
3941   /Details/170243/perbup-kab-siak-no-15-tahun-2020   
3942  /Details/252594/pergub-prov-jawa-timur-no-67-t...   
3943  /Details/165332/perbup-kab-katingan-no-15-tahu...   
3944   /Details/170533/perbup-kab-siak-no-27-tahun-2020   

                                             href_unduh  
0          /Download/38966/PP%20NO%2052%20TH%202005.pdf  
1          /Download/38952/PP%20NO%2050%20TH%202005.pdf  
2          /Download/38955/PP%20NO%2051%20TH%202005.pdf  
3          /Download/38618/PP%20NO%2011%20TH%202005.pdf  
4     /Download/254819/Permen%20Kominfo%20Nomor%2043...  
...             

In [109]:
from google.colab import files
regulasi_df.to_csv('daftar_regulasi_penyiaran_daerah_01042024.csv')
files.download('daftar_regulasi_penyiaran_daerah_01042024.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [7]:
meta_regulasi_df = pd.DataFrame(meta_regulasi)
meta_regulasi_df

judul_panjang tahun  \
0     Peraturan Pemerintah (PP) tentang Penyelenggar...  2005   
1     Peraturan Pemerintah (PP) tentang Penyelenggar...  2005   
2     Peraturan Pemerintah (PP) tentang Penyelenggar...  2005   
3     Peraturan Pemerintah (PP) tentang Penyelenggar...  2005   
4     Peraturan Menteri Komunikasi dan Informatika t...  2009   
...                                                 ...   ...   
3940  Peraturan Bupati (PERBUP) tentang Perubahan Ke...  2020   
3941  Peraturan Bupati (PERBUP) tentang Perubahan Ke...  2020   
3942  Peraturan Gubernur (PERGUB) tentang Perubahan ...  2022   
3943  Peraturan Bupati (PERBUP) tentang Penggolongan...  2020   
3944  Peraturan Bupati (PERBUP) tentang Perubahan Ke...  2020   

                                                 sumber  \
0     Peraturan Pemerintah (PP) NO. 52, LN. 2005 No....   
1     Peraturan Pemerintah (PP) NO. 50, LN. 2005 No....   
2     Peraturan Pemerintah (PP) NO. 51, LN. 2005 No....   
3     Peraturan Pemerintah (PP) NO. 11, LN. 2005 No....   
4     Peraturan Menteri Komunikasi dan Informatika N...   
...                                                 ...   
3940  Peraturan Bupati (PERBUP) NO. 26, Berita Daera...   
3941  Peraturan Bupati (PERBUP) NO. 15, Berita Daera...   
3942  Peraturan Gubernur (PERGUB) NO. 67, Berita Dae...   
3943      Peraturan Bupati (PERBUP) NO. 15, BD.2020/554   
3944  Peraturan Bupati (PERBUP) NO. 27, Berita Daera...   

                                          judul_singkat  \
0                                                         
1                                                         
2                                                         
3                                                         
4                                                         
...                                                 ...   
3940  Perubahan Kedelapan atas Peraturan Bupati Siak...   
3941  PERUBAHAN KETIGA ATAS PERATURAN BUPATI SIAK NO...   
3942                                                SBU   
3943                                      Barang Daerah   
3944  Perubahan Kesembilan atas Peraturan Bupati Sia...   

                                           pertimbangan  \
0                                                         
1                                                         
2                                                         
3                                                         
4                                                         
...                                                 ...   
3940  a. bahwa berdasarkan Lampiran Peraturan Menter...   
3941  Bahwa untuk melaksanakan ketentuan Pasal 15 ay...   
3942  Menimbang: a. bahwa untuk optimalisasi dan kel...   
3943  a.  bahwa untuk melaksanakan  Ketentuan  Pasal...   
3944  a. bahwa berdasarkan Peraturan Menteri Keuanga...   

                                             konsideran  \
0                                                         
1                                                         
2                                                         
3                                                         
4                                                         
...                                                 ...   
3940  Dasar Hukum Perbup ini adalah: Pasal 18 Ayat (...   
3941  Dasar Hukum Perbup ini adalah: Pasal 18 ayat (...   
3942  Mengingat: 1. Pasal 18 ayat (6) Undang-Undang ...   
3943  Undang  -  Undang  Nomor  5  Tahun  2002  tent...   
3944  Dasar Hukum Perbup ini adalah: Pasal 18 Ayat (...   

                                           materi_pokok  \
0                                                         
1                                                         
2                                                         
3                                                         
4                                                         
...                                                 ...   
3940  Dalam 

In [ ]:
meta_regulasi_df.to_csv('meta_regulasi_penyiaran_daerah_01042024.csv')
files.download('meta_regulasi_penyiaran_daerah_01042024.csv')

In [125]:
data_df = pd.DataFrame(result[0]['data'])
data_df

data_id                                           jenis_no  \
0        194  Peraturan Daerah (PERDA) Provinsi Daerah Istim...   
1     289263  Peraturan Daerah (PERDA) Kabupaten Kendal Nomo...   
2     321871  Peraturan Daerah (PERDA) Kabupaten Grobogan No...   
3     302328  Peraturan Bupati (PERBUP) Kabupaten Kudus Nomo...   
4      58173  Peraturan Daerah (PERDA) Provinsi Gorontalo No...   
...      ...                                                ...   
2916  228833  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
2917  228813  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
2918  309224  Peraturan Gubernur (PERGUB) Provinsi Jawa Timu...   
2919  158222  Peraturan Bupati (PERBUP) Kabupaten Katingan N...   
2920  228834  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   

                                                  judul  \
0                             Penyelenggaraan Penyiaran   
1     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...   
2     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...   
3     Lembaga Penyiaran Publik Lokal Jasa Penyiaran ...   
4                 Penyelenggaran Penyiaran Berlangganan   
...                                                 ...   
2916  Perubahan Kedelapan atas Peraturan Bupati Siak...   
2917  Perubahan Ketiga atas Peraturan Bupati Siak No...   
2918  Perubahan atas Peraturan Gubernur Nomor 32 Tah...   
2919  Penggolongan dan kodifikasi Barang Milik Daera...   
2920  Perubahan Kesembilan atas Peraturan Bupati Sia...   

                                                 subjek  \
0                  Informasi, Pers, Pos, dan Periklanan   
1       Informasi, Pers, Pos, dan Periklanan|Mencabut :   
2      Telekomunikasi, Informatika, Siber, dan Internet   
3     Telekomunikasi, Informatika, dan Internet|Dasa...   
4                  Informasi, Pers, Pos, dan Periklanan   
...                                                 ...   
2916  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   
2917  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   
2918                               APBD|Standar/Pedoman   
2919             Pengelolaan Barang Milik Negara/Daerah   
2920  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   

                                               cuplikan  \
0     ... Lembaga Penyiaran Publik Lokal.  Paragraf ...   
1     ... Bagian Kesepuluh J asa Tambahan Penyiaran ...   
2     ... 1 PEMERINTAH   KABUPATEN   GROBOGAN   PERA...   
3     ... Lembaga Penyiaran Publik Lokal yang selanj...   
4     ... Izin Penyelenggaraan Penyiaran adalah hak ...   
...                                                 ...   
2916  ... Bus Air 613.152 000.00 613.152.000.00 O.00...   
2917  ... Bus Air 613.152,000.00 613.152.000.00 6.00...   
2918  ... Kegiatan 50.000.000       2231 8.1.02.02.0...   
2919  ... STATISTIK 62 6 2-16 0-0 0-0 02 SEKRETARIAT...   
2920  ... Programi 7.575.392.000.00 7.575.302.000,00...   

                                             href_detil  \
0        /Details/11591/perda-prov-diy-no-13-tahun-2016   
1      /Details/235555/perda-kab-kendal-no-7-tahun-2006   
2     /Details/264728/perda-kab-grobogan-no-2-tahun-...   
3      /Details/246219/perbup-kab-kudus-no-6-tahun-2023   
4     /Details/67816/perda-prov-gorontalo-no-15-tahu...   
...                                                 ...   
2916   /Details/170256/perbup-kab-siak-no-26-tahun-2020   
2917   /Details/170243/perbup-kab-siak-no-15-tahun-2020   
2918  /Details/252594/pergub-prov-jawa-timur-no-67-t...   
2919  /Details/165332/perbup-kab-katingan-no-15-tahu...   
2920   /Details/170533/perbup-kab-siak-no-27-tahun-2020   

                                             href_unduh  
0     /Download/194/Perda%20DIY%20No.%2013%20Tahun%2...  
1     /Download/289263/PERDA%20NO.%207%20TAHUN%20200...  
2     /Download/321871/Perda%20Grobogan%202%20Tahun%...  
3          /Download/302328/Perbup%20Kudus%206_2023.pdf  
4     /Download/58173/Perda%20Prov%20Gorontalo%20No%...  
...             

In [123]:
meta_df = pd.DataFrame(result[0]['meta'])
meta_df

judul_panjang tahun  \
0     Peraturan Daerah (PERDA) tentang Penyelenggara...  2016   
1     Peraturan Daerah (PERDA) tentang Penyelenggara...  2006   
2     Peraturan Daerah (PERDA) tentang Penyelenggara...  2008   
3     Peraturan Bupati (PERBUP) tentang Lembaga Peny...  2023   
4     Peraturan Daerah (PERDA) tentang Penyelenggara...  2015   
...                                                 ...   ...   
2916  Peraturan Bupati (PERBUP) tentang Perubahan Ke...  2020   
2917  Peraturan Bupati (PERBUP) tentang Perubahan Ke...  2020   
2918  Peraturan Gubernur (PERGUB) tentang Perubahan ...  2022   
2919  Peraturan Bupati (PERBUP) tentang Penggolongan...  2020   
2920  Peraturan Bupati (PERBUP) tentang Perubahan Ke...  2020   

                                                 sumber  \
0        Peraturan Daerah (PERDA) NO. 13, LD.2016/NO.13   
1          Peraturan Daerah (PERDA) NO. 7, LD.2006/No.7   
2     Peraturan Daerah (PERDA) NO. 2, LD.2008/No.1 S...   
3         Peraturan Bupati (PERBUP) NO. 6, BD.2023/NO.6   
4        Peraturan Daerah (PERDA) NO. 15, LD.2015/NO.15   
...                                                 ...   
2916  Peraturan Bupati (PERBUP) NO. 26, Berita Daera...   
2917  Peraturan Bupati (PERBUP) NO. 15, Berita Daera...   
2918  Peraturan Gubernur (PERGUB) NO. 67, Berita Dae...   
2919      Peraturan Bupati (PERBUP) NO. 15, BD.2020/554   
2920  Peraturan Bupati (PERBUP) NO. 27, Berita Daera...   

                                          judul_singkat  \
0                                             Penyiaran   
1                             PENYELENGGARAAN PENYIARAN   
2     penyelenggaraan-lembaga-penyiaran-publik-lokal...   
3                              lembaga penyiaran publik   
4     PENYELENGGARAAN PENYIARAN BERLANGGANAN TELEVIS...   
...                                                 ...   
2916  Perubahan Kedelapan atas Peraturan Bupati Siak...   
2917  PERUBAHAN KETIGA ATAS PERATURAN BUPATI SIAK NO...   
2918                                                SBU   
2919                                      Barang Daerah   
2920  Perubahan Kesembilan atas Peraturan Bupati Sia...   

                                           pertimbangan  \
0     bahwa penyelenggaraan penyiaran di daerah meru...   
1     bahwa keberadaan radio sebagai media penyiaran...   
2     a. bahwa keberadaan radio sebagai media penyia...   
3     bahwa dalam rangka melaksanakan Ketentuan Pasa...   
4     Peraturan ini dibentuk dalam rangka memperoleh...   
...                                                 ...   
2916  a. bahwa berdasarkan Lampiran Peraturan Menter...   
2917  Bahwa untuk melaksanakan ketentuan Pasal 15 ay...   
2918  Menimbang: a. bahwa untuk optimalisasi dan kel...   
2919  a.  bahwa untuk melaksanakan  Ketentuan  Pasal...   
2920  a. bahwa berdasarkan Peraturan Menteri Keuanga...   

                                             konsideran  \
0     Dasar Hukum : Pasal 18 ayat (6) Undang-Undang ...   
1     Undang-undang Nomor 13 Tahun 1950; Undang-unda...   
2     Undang-Undang Nomor 13 Tahun 1950; Undang-Unda...   
3     Undang-Undang Nomor 13 Tahun 1950; Undang-Unda...   
4     Dasar hukum Peraturan Daerah ini adalah Pasal ...   
...                                                 ...   
2916  Dasar Hukum Perbup ini adalah: Pasal 18 Ayat (...   
2917  Dasar Hukum Perbup ini adalah: Pasal 18 ayat (...   
2918  Mengingat: 1. Pasal 18 ayat (6) Undang-Undang ...   
2919  Undang  -  Undang  Nomor  5  Tahun  2002  tent...   
2920  Dasar Hukum Perbup ini adalah: Pasal 18 Ayat (...   

                                           materi_pokok  \
0     Materi Pokok: Kelembagaan Penyiaran, Kepemilik...   
1     Peraturan Daerah ini mengatur tentang ketentua...   
2     Peraturan ini mengatur Penyelenggaraan kegiata...   
3     Di dalam Peraturan Bupati ini diatur tentang:\...   
4     Dalam peraturan ini diatur tentang Penyelengga...   
...                                                 ...   
2916  Dalam 

In [131]:
policy_df = pd.concat([data_df, meta_df], axis=1)
policy_df

data_id                                           jenis_no  \
0        194  Peraturan Daerah (PERDA) Provinsi Daerah Istim...   
1     289263  Peraturan Daerah (PERDA) Kabupaten Kendal Nomo...   
2     321871  Peraturan Daerah (PERDA) Kabupaten Grobogan No...   
3     302328  Peraturan Bupati (PERBUP) Kabupaten Kudus Nomo...   
4      58173  Peraturan Daerah (PERDA) Provinsi Gorontalo No...   
...      ...                                                ...   
2916  228833  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
2917  228813  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
2918  309224  Peraturan Gubernur (PERGUB) Provinsi Jawa Timu...   
2919  158222  Peraturan Bupati (PERBUP) Kabupaten Katingan N...   
2920  228834  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   

                                                  judul  \
0                             Penyelenggaraan Penyiaran   
1     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...   
2     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...   
3     Lembaga Penyiaran Publik Lokal Jasa Penyiaran ...   
4                 Penyelenggaran Penyiaran Berlangganan   
...                                                 ...   
2916  Perubahan Kedelapan atas Peraturan Bupati Siak...   
2917  Perubahan Ketiga atas Peraturan Bupati Siak No...   
2918  Perubahan atas Peraturan Gubernur Nomor 32 Tah...   
2919  Penggolongan dan kodifikasi Barang Milik Daera...   
2920  Perubahan Kesembilan atas Peraturan Bupati Sia...   

                                                 subjek  \
0                  Informasi, Pers, Pos, dan Periklanan   
1       Informasi, Pers, Pos, dan Periklanan|Mencabut :   
2      Telekomunikasi, Informatika, Siber, dan Internet   
3     Telekomunikasi, Informatika, dan Internet|Dasa...   
4                  Informasi, Pers, Pos, dan Periklanan   
...                                                 ...   
2916  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   
2917  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   
2918                               APBD|Standar/Pedoman   
2919             Pengelolaan Barang Milik Negara/Daerah   
2920  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   

                                               cuplikan  \
0     ... Lembaga Penyiaran Publik Lokal.  Paragraf ...   
1     ... Bagian Kesepuluh J asa Tambahan Penyiaran ...   
2     ... 1 PEMERINTAH   KABUPATEN   GROBOGAN   PERA...   
3     ... Lembaga Penyiaran Publik Lokal yang selanj...   
4     ... Izin Penyelenggaraan Penyiaran adalah hak ...   
...                                                 ...   
2916  ... Bus Air 613.152 000.00 613.152.000.00 O.00...   
2917  ... Bus Air 613.152,000.00 613.152.000.00 6.00...   
2918  ... Kegiatan 50.000.000       2231 8.1.02.02.0...   
2919  ... STATISTIK 62 6 2-16 0-0 0-0 02 SEKRETARIAT...   
2920  ... Programi 7.575.392.000.00 7.575.302.000,00...   

                                             href_detil  \
0        /Details/11591/perda-prov-diy-no-13-tahun-2016   
1      /Details/235555/perda-kab-kendal-no-7-tahun-2006   
2     /Details/264728/perda-kab-grobogan-no-2-tahun-...   
3      /Details/246219/perbup-kab-kudus-no-6-tahun-2023   
4     /Details/67816/perda-prov-gorontalo-no-15-tahu...   
...                                                 ...   
2916   /Details/170256/perbup-kab-siak-no-26-tahun-2020   
2917   /Details/170243/perbup-kab-siak-no-15-tahun-2020   
2918  /Details/252594/pergub-prov-jawa-timur-no-67-t...   
2919  /Details/165332/perbup-kab-katingan-no-15-tahu...   
2920   /Details/170533/perbup-kab-siak-no-27-tahun-2020   

                                             href_unduh  \
0     /Download/194/Perda%20DIY%20No.%2013%20Tahun%2...   
1     /Download/289263/PERDA%20NO.%207%20TAHUN%20200...   
2     /Download/321871/Perda%20Grobogan%202%20Tahun%...   
3          /Download/302328/Perbup%20Kudus%206_2023.pdf   
4     /Download/58173/Perda%20Prov%20Gorontalo%20No%...   
...       

In [ ]:
policy_df.to_csv('regulasi_penyiaran_daerah_merged_01042024.csv')
files.download('regulasi_penyiaran_daerah_merged_01042024.csv')

In [10]:
broadcast_policies = pd.concat([regulasi_df, meta_regulasi_df], axis=1)
broadcast_policies

data_id                                           jenis_no  \
0      38966        Peraturan Pemerintah (PP) No. 52 Tahun 2005   
1      38952        Peraturan Pemerintah (PP) No. 50 Tahun 2005   
2      38955        Peraturan Pemerintah (PP) No. 51 Tahun 2005   
3      38618        Peraturan Pemerintah (PP) No. 11 Tahun 2005   
4     254819  Peraturan Menteri Komunikasi dan Informatika N...   
...      ...                                                ...   
3940  228833  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
3941  228813  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
3942  309224  Peraturan Gubernur (PERGUB) Provinsi Jawa Timu...   
3943  158222  Peraturan Bupati (PERBUP) Kabupaten Katingan N...   
3944  228834  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   

                                                  judul  \
0     Penyelenggaraan Penyiaran Lembaga Penyiaran Be...   
1     Penyelenggaraan Penyiaran Lembaga Penyiaran Sw...   
2     Penyelenggaraan Penyiaran Lembaga Penyiaran Ko...   
3     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...   
4     Penyelenggaraan Penyiaran Melalui Sistem Stasi...   
...                                                 ...   
3940  Perubahan Kedelapan atas Peraturan Bupati Siak...   
3941  Perubahan Ketiga atas Peraturan Bupati Siak No...   
3942  Perubahan atas Peraturan Gubernur Nomor 32 Tah...   
3943  Penggolongan dan kodifikasi Barang Milik Daera...   
3944  Perubahan Kesembilan atas Peraturan Bupati Sia...   

                                                 subjek  \
0     Informasi, Pers, Pos, dan Periklanan|Diubah de...   
1     Informasi, Pers, Pos, dan Periklanan|Diubah de...   
2     Informasi, Pers, Pos, dan Periklanan|Diubah de...   
3     Informasi, Pers, Pos, dan Periklanan|Diubah de...   
4     Telekomunikasi, Informatika, dan Internet|Dica...   
...                                                 ...   
3940  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   
3941  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   
3942                               APBD|Standar/Pedoman   
3943             Pengelolaan Barang Milik Negara/Daerah   
3944  APBD|Pengelolaan Keuangan Negara/Daerah|Mengub...   

                                               cuplikan  \
0     ... Peraturan Pemerintah tentang Penyelenggara...   
1     ... Peraturan Pemerintah ini telah memiliki Le...   
2     ... BAB V   RENCANA DASAR TEKNIK DAN PERSYARAT...   
3     ... Siaran, penyiaran, penyiaran radio, penyia...   
4     ... 1 Menimbang : a. bahwa penyiaran diselengg...   
...                                                 ...   
3940  ... Bus Air 613.152 000.00 613.152.000.00 O.00...   
3941  ... Bus Air 613.152,000.00 613.152.000.00 6.00...   
3942  ... Kegiatan 50.000.000       2231 8.1.02.02.0...   
3943  ... STATISTIK 62 6 2-16 0-0 0-0 02 SEKRETARIAT...   
3944  ... Programi 7.575.392.000.00 7.575.302.000,00...   

                                             href_detil  \
0                    /Details/49688/pp-no-52-tahun-2005   
1                    /Details/49674/pp-no-50-tahun-2005   
2                    /Details/49678/pp-no-51-tahun-2005   
3                    /Details/49339/pp-no-11-tahun-2005   
4     /Details/159041/permenkominfo-no-43permkominfo...   
...                                                 ...   
3940   /Details/170256/perbup-kab-siak-no-26-tahun-2020   
3941   /Details/170243/perbup-kab-siak-no-15-tahun-2020   
3942  /Details/252594/pergub-prov-jawa-timur-no-67-t...   
3943  /Details/165332/perbup-kab-katingan-no-15-tahu...   
3944   /Details/170533/perbup-kab-siak-no-27-tahun-2020   

                                             href_unduh  \
0          /Download/38966/PP%20NO%2052%20TH%202005.pdf   
1          /Download/38952/PP%20NO%2050%20TH%202005.pdf   
2          /Download/38955/PP%20NO%2051%20TH%202005.pdf   
3          /Download/38618/PP%20NO%2011%20TH%202005.pdf   
4     /Download/254819/Permen%20Kominfo%20Nomor%2043...   
...       

In [11]:
broadcast_policies.to_csv('data/regulasi_penyiaran_id.csv', index=False)

Trial and Error

In [83]:
url = 'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor=&jenis=19&jenis=20&jenis=23'
r1 = s.get(url, headers=headers)
res1 = soup(r1.text, 'html.parser')

In [ ]:

# for item in res1.find_all('div', attrs={'class':'flex-grow-1'}):
for item in res1.find('div', attrs={'class':'rounded-4'}).find_all('div', attrs={'class':'row mb-8'}):


  # print(item.find('div', attrs={'class':'col-lg-8 fw-bold fs-4'}).text.strip() if item.find('div', attrs={'class':'col-lg-8 fw-bold fs-4'}) != None else '') # jenis dan nomor peraturan
  # print(item.find('a', attrs={'class':'fs-1 fw-bold'})['href'] if item.find('a', attrs={'class':'fs-1 fw-bold'}) != None else '') # link page_details https://peraturan.bpk.go.id/{page_details}
  # print(item.find('a', attrs={'class':'fs-1 fw-bold'}).text.strip().replace('\r\n', ' ')) # Nama /Judul Peraturan
  # print('|'.join([x.text for x in item.find_all('span', attrs={'class':'badge'})])) # subjek peraturan
  # print('|'.join([a.text for a in item.find_all('div', attrs={'class':'mb-8'})])) # cuplikan text
  # print(item.find('a', attrs={'class':'download-file'})['href']) # href download
  # print(item.find('a', attrs={'class':'download-file'})['data-id']) # data-id
  # print(item.find('a', attrs={'class':'download-file'}).text.strip()) # nama file



In [4]:
len(res.find_all('div', attrs={'class':'modal-body'}))

11

In [ ]:
res.find_all('div', attrs={'class':'modal-body'})

In [36]:
for abstrak in res.find_all('div', attrs={'class':'modal-body'}):
  if abstrak.find('div', attrs={'class':'fs-5'}) != None:
    judul_singkat = abstrak.find('div', attrs={'class':'fs-5'}).find_all('p')[0].text.strip() # judul singkat
    tahun = abstrak.find('div', attrs={'class':'fs-5'}).find_all('p')[1].text.strip() # tahun
    sumber = abstrak.find('div', attrs={'class':'fs-5'}).find_all('p')[2].text.strip() # sumber
    judul_lengkap = abstrak.find('div', attrs={'class':'fs-5'}).find_all('p')[3].text.strip().replace('\r\n', ' ') # judul lengkap

    # print(len(abstrak.find_all('tr')[0].find_all('li')))

    pertimbangan = abstrak.find_all('tr')[0].find_all('li')[0].text.strip() if len(abstrak.find_all('tr')[0].find_all('li')) == 3 else '' # pertimbangan
    konsideran = abstrak.find_all('tr')[0].find_all('li')[1].text.strip() if len(abstrak.find_all('tr')[0].find_all('li')) == 3 else '' # konsideran
    materi_pokok = abstrak.find_all('tr')[0].find_all('li')[2].text.replace('\r\n','').strip() if len(abstrak.find_all('tr')[0].find_all('li')) == 3 else '' # materi pokok

    catatan = '\n'.join([x.text for x in abstrak.find_all('tr')[1].find_all('li')]) # catatan

    print(pertimbangan, konsideran, materi_pokok, catatan)

bahwa penyelenggaraan penyiaran di daerah merupakan kebutuhan untuk aktualisasi nilai-nilai kebudayaan/kearifan lokal Daerah Istimewa Yogyakarta sekaligus sebagai upaya untuk mencerdaskan kehidupan masyarakat dengan informasi yang bermanfaat Dasar Hukum : Pasal 18 ayat (6) Undang-Undang Dasar Negara Republik Indonesia Tahun 1945, Undang-Undang Nomor 3 Tahun 1950 tentang Pembentukan Daerah Istimewa Jogjakarta (Berita Negara Republik Indonesia Tahun 1950 Nomor 3), sebagaimana telah diubah terakhir dengan Undang-Undang Nomor 9 Tahun 1955
tentang Perubahan Undang-Undang Nomor 3 Jo. Nomor 19 Tahun 1950 tentang Pembentukan Daerah Istimewa, Undang-Undang Nomor 32 Tahun 2002 tentang Penyiaran, Undang-Undang Nomor 13 Tahun 2012 tentang Keistimewaan Daerah Istimewa Yogyakarta, Undang-Undang Nomor 23 Tahun 2014 tentang Pemerintahan
Daerah sebagaimana telah diubah terakhir dengan Undang-Undang Nomor 9 Tahun 2015 tentang Perubahan Kedua Atas Undang-Undang Nomor 23 Tahun 2014 tentang Pemerintahan Da

In [ ]:
# https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor=&jenis=19&jenis=20&p=2

## Mengambil detail data

In [29]:
metadata_regulasi = []
riwayat_regulasi = []
riwayat_uji_materi = []

url = 'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor='
r1 = s.get(url, headers=headers)
res1 = soup(r1.text, 'html.parser')

last_pages = int(res1.find('ul', attrs={'class':'pagination'}).find_all('a', attrs={'class':'page-link'})[-1]['href'].split('=')[-1]) # jumlah halaman
pages = last_pages + 1

for page_number in tqdm(range(1, pages)):
  url_ = f'https://peraturan.bpk.go.id/Search?keywords=penyiaran&tentang=&nomor=&p={page_number}'
  r2 = s.get(url_, headers=headers)
  res2 = soup(r2.text, 'html.parser')

  for item in res2.find('div', attrs={'class':'rounded-4'}).find_all('div', attrs={'class':'row mb-8'}):
    href_detil = item.find('a', attrs={'class':'fs-1 fw-bold'})['href']
    url_detil = f'https://peraturan.bpk.go.id{href_detil}'

    r = s.get(url_detil, headers=headers)
    if r.status_code==200:
      res = soup(r.text, 'html.parser')

      data_id = res.find_all('div', attrs={'class':'card-body'})[2].find_all('a')[-1]['data-id']
      judul_singkat = res.find('title').text.strip()
      judul_panjang = res.find('h4').text.strip()
      judul_tentang = res.find('h1').text.strip()
      materi_pokok = res.find_all('div', attrs={'class':'card-body'})[0].find('p').text.strip() if res.find_all('div', attrs={'class':'card-body'})[0].find('p') != None else ''
      href_unduh = res.find_all('div', attrs={'class':'card-body'})[2].find_all('a')[-1]['href'] # Link File Peraturan
      link_unduh = f'https://peraturan.bpk.go.id{href_unduh}'

      for meta in res.find_all('div', attrs={'class':'card-body'})[1].find_all('div', attrs={'class':'container'}):
        metadata_regulasi.append({
          'data_id': data_id,
          'judul_singkat': judul_singkat,
          'judul_panjang': judul_panjang,
          'judul_tentang': judul_tentang,
          'judul_lengkap': meta.find_all('div', attrs={'class':'col-lg-9'})[1].text.strip(),
          'materi_pokok': materi_pokok,
          'tipe_dokumen': meta.find_all('div', attrs={'class':'col-lg-9'})[0].text.strip(),
          'teu': meta.find_all('div', attrs={'class':'col-lg-9'})[2].text.strip(),
          'nomor' : meta.find_all('div', attrs={'class':'col-lg-9'})[3].text.strip(),
          'bentuk' : meta.find_all('div', attrs={'class':'col-lg-9'})[4].text.strip(),
          'bentuk_singkat' : meta.find_all('div', attrs={'class':'col-lg-9'})[5].text.strip(),
          'tahun' : meta.find_all('div', attrs={'class':'col-lg-9'})[6].text.strip(),
          'tempat_penetapan' : meta.find_all('div', attrs={'class':'col-lg-9'})[7].text.strip(),
          'tgl_penetapan' : meta.find_all('div', attrs={'class':'col-lg-9'})[8].text.strip(),
          'tgl_pengundangan' : meta.find_all('div', attrs={'class':'col-lg-9'})[9].text.strip(),
          'tgl_berlaku' : meta.find_all('div', attrs={'class':'col-lg-9'})[10].text.strip(),
          'sumber' : meta.find_all('div', attrs={'class':'col-lg-9'})[11].text.strip(),
          'subjek' : meta.find_all('div', attrs={'class':'col-lg-9'})[12].text.strip(),
          'status' : meta.find_all('div', attrs={'class':'col-lg-9'})[13].text.strip(),
          'bahasa' : meta.find_all('div', attrs={'class':'col-lg-9'})[14].text.strip(),
          'lokasi' : meta.find_all('div', attrs={'class':'col-lg-9'})[15].text.strip(),
          'bidang' : meta.find_all('div', attrs={'class':'col-lg-9'})[16].text.strip(),
          'jml_akses' : meta.find('div', attrs={'class':'col-lg-6'}).text.strip().split(' ')[-2],
          'link_unduh': link_unduh
        })

      # RIWAYAT PERATURAN
      riwayat = res.find_all('div', attrs={'class':'card-body'})[3]

      if riwayat.find('div', attrs={'class':'text-danger'}) != None:
        relasi_peraturan = ''
        rujukan_regulasi = ''
        rujukan_tentang = ''

      else:

        # for relasi in riwayat.find_all('div', attrs={'class':'row mb-4'}):
        for relasi in riwayat.find_all('div', attrs={'class':'row'}):
          if relasi.find('div', attrs={'class':'p-4'}) != None:
            relasi_peraturan = relasi.find('div', attrs={'class':'p-4'}).text.strip()

          for rujukan in relasi.find_all('li'):
            if rujukan.find('a') != None:
              regulasi_rujukan = rujukan.a.text.strip()
              regulasi_tentang = rujukan.text.strip().split('\n')[1].replace('tentang ', '')
            else:
              regulasi_rujukan = rujukan.text.strip().replace('Nomor', 'No.').split(' tentang ')[0]
              regulasi_tentang = rujukan.text.strip().replace('Nomor', 'No.').split(' tentang ')[1] if len(rujukan.text.strip().replace('Nomor', 'No.').split(' tentang ')) > 1 else ''

            data_riwayat = {'data_id': data_id, 'judul_singkat': judul_singkat, 
                            'judul_tentang': judul_tentang, 'relasi_peraturan': relasi_peraturan, 
                            'regulasi_rujukan': regulasi_rujukan, 'regulasi_tentang':regulasi_tentang}

            riwayat_regulasi.append(data_riwayat)

      ## UJI MATERI MK
      uji_materi = res.find_all('div', attrs={'class':'card-body'})[4]

      if uji_materi.find('div', attrs={'class':'text-danger'}) != None:
        puu_mk_no = ''
        puu_mk_ringkas = ''

      else:
        for uji in uji_materi.find_all('div', attrs={'class':'row g-4 g-xl-9 mb-8'}):
          puu_mk_no = uji.find_all('div', attrs={'class':'col-12'})[0].text.strip()
          puu_mk_ringkas = uji.find_all('div', attrs={'class':'col-12'})[1].text.strip()

          riwayat_mk = {'data_id': data_id, 'judul_singkat': judul_singkat, 'judul_tentang': judul_tentang,
                        'puu_mk_no': puu_mk_no, 'puu_mk_ringkas': puu_mk_ringkas}
          
          riwayat_uji_materi.append(riwayat_mk)
    else:
      pass

100%|██████████| 395/395 [33:57<00:00,  5.16s/it]  


In [31]:
metadata_regulasi_df = pd.DataFrame(metadata_regulasi)
metadata_regulasi_df

data_id                                      judul_singkat  \
0      38966                               PP No. 52 Tahun 2005   
1      38952                               PP No. 50 Tahun 2005   
2      38955                               PP No. 51 Tahun 2005   
3      38618                               PP No. 11 Tahun 2005   
4     254819  Permenkominfo No. 43/PER/M.KOMINFO/10/2009 Tah...   
...      ...                                                ...   
3939  228833                 PERBUP Kab. Siak No. 26 Tahun 2020   
3940  228813                 PERBUP Kab. Siak No. 15 Tahun 2020   
3941  309224          PERGUB Prov. Jawa Timur No. 67 Tahun 2022   
3942  158222             PERBUP Kab. Katingan No. 15 Tahun 2020   
3943  228834                 PERBUP Kab. Siak No. 27 Tahun 2020   

                                          judul_panjang  \
0           Peraturan Pemerintah (PP) No. 52 Tahun 2005   
1           Peraturan Pemerintah (PP) No. 50 Tahun 2005   
2           Peraturan Pemerintah (PP) No. 51 Tahun 2005   
3           Peraturan Pemerintah (PP) No. 11 Tahun 2005   
4     Peraturan Menteri Komunikasi dan Informatika N...   
...                                                 ...   
3939  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
3940  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
3941  Peraturan Gubernur (PERGUB) Provinsi Jawa Timu...   
3942  Peraturan Bupati (PERBUP) Kabupaten Katingan N...   
3943  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   

                                          judul_tentang  \
0     Penyelenggaraan Penyiaran Lembaga Penyiaran Be...   
1     Penyelenggaraan Penyiaran Lembaga Penyiaran Sw...   
2     Penyelenggaraan Penyiaran Lembaga Penyiaran Ko...   
3     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...   
4     Penyelenggaraan Penyiaran Melalui Sistem Stasi...   
...                                                 ...   
3939  Perubahan Kedelapan atas Peraturan Bupati Siak...   
3940  Perubahan Ketiga atas Peraturan Bupati Siak No...   
3941  Perubahan atas Peraturan Gubernur Nomor 32 Tah...   
3942  Penggolongan dan kodifikasi Barang Milik Daera...   
3943  Perubahan Kesembilan atas Peraturan Bupati Sia...   

                                          judul_lengkap  \
0     Peraturan Pemerintah (PP)  Nomor 52 Tahun 2005...   
1     Peraturan Pemerintah (PP)  Nomor 50 Tahun 2005...   
2     Peraturan Pemerintah (PP)  Nomor 51 Tahun 2005...   
3     Peraturan Pemerintah (PP)  Nomor 11 Tahun 2005...   
4     Peraturan Menteri Komunikasi dan Informatika  ...   
...                                                 ...   
3939  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
3940  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   
3941  Peraturan Gubernur (PERGUB) Provinsi Jawa Timu...   
3942  Peraturan Bupati (PERBUP) Kabupaten Katingan N...   
3943  Peraturan Bupati (PERBUP) Kabupaten Siak Nomor...   

                                           materi_pokok  \
0                                                         
1                                                         
2                                                         
3                                                         
4                                                         
...                                                 ...   
3939  Dalam Peraturan Bupati ini berisi 3 (tiga) Bab...   
3940  Dalam Peraturan Bupati ini berisi 3 (tiga) Bab...   
3941  mengatur tentang Perubahan ketentuan dalam Lam...   
3942  a.  Kodefikasi Barang;\r\nb.  Kode Lokasi; dan...   
3943  Dalam Peraturan Bupati ini berisi 3 (tiga) Bab...   

                      tipe_dokumen  \
0     Peraturan Perundang-undangan   
1     Peraturan Perundang-undangan   
2     Peraturan Perundang-undangan   
3     Peraturan Perundang-undangan   
4     Peraturan Perundang-undangan   
...                            ...   
3939  Peraturan Perundang-undangan   
3940  Peraturan Perundang-undangan   
3941  Peraturan Perund

Secara keseluruhan, jumlah regulasi yang mengandung kata 'penyiaran' pada database peraturan.bpk.go.id sejumlah 3945 entri. Namun ada 1 entri di mana laman detail regulasi mengalami galat. Dengan demikian jumlah entri regulasi terkait penyiaran yang berhasil didapatkan sejumlah 3944 entri.

In [58]:
metadata_regulasi_df['status'].value_counts()

Berlaku          3533
Tidak Berlaku     411
Name: status, dtype: int64

In [59]:
metadata_regulasi_df['bentuk'].value_counts()

Peraturan Daerah (PERDA)                                 1384
Peraturan Bupati (PERBUP)                                1249
Peraturan Walikota (PERWALI)                              297
Peraturan Gubernur (PERGUB)                               288
Peraturan Menteri Komunikasi dan Informatika              137
                                                         ... 
Peraturan Perpustakaan Nasional                             1
Peraturan Menteri Pekerjaan Umum dan Perumahan Rakyat       1
Peraturan Arsip Nasional Republik Indonesia                 1
Peraturan Badan Narkotika Nasional                          1
Peraturan Badan Pengawas Obat dan Makanan                   1
Name: bentuk, Length: 62, dtype: int64

In [60]:
metadata_regulasi_df['bentuk'].unique()

array(['Peraturan Pemerintah (PP)',
       'Peraturan Menteri Komunikasi dan Informatika',
       'Undang-undang (UU)', 'Peraturan Daerah (PERDA)',
       'Keputusan Presiden (Keppres)', 'Peraturan Bupati (PERBUP)',
       'Peraturan Walikota (PERWALI)', 'Peraturan Gubernur (PERGUB)',
       'Qanun', 'Peraturan Presiden (Perpres)',
       'Peraturan Menteri Keuangan',
       'Peraturan Kepala Badan Kepegawaian Negara',
       'Peraturan Badan Pengawas Pemilihan Umum',
       'Peraturan Menteri Pendidikan dan Kebudayaan',
       'Peraturan Komisi Pemilihan Umum',
       'Peraturan Kepala Badan Koordinasi Penanaman Modal',
       'Peraturan Badan Pemeriksa Keuangan',
       'Keputusan Badan Pemeriksa Keuangan',
       'Peraturan Badan Narkotika Nasional',
       'Peraturan Menteri Dalam Negeri',
       'Peraturan Menteri Pendayagunaan Aparatur Negara dan Reformasi Birokrasi',
       'Instruksi Presiden (Inpres)',
       'Peraturan Pemerintah Pengganti Undang-Undang (Perpu)',
       'Pera

In [61]:
metadata_regulasi_df['judul_tentang'].unique()

array(['Penyelenggaraan Penyiaran Lembaga Penyiaran Berlangganan',
       'Penyelenggaraan Penyiaran Lembaga Penyiaran Swasta',
       'Penyelenggaraan Penyiaran Lembaga Penyiaran Komunitas', ...,
       'Perubahan atas Peraturan Gubernur Nomor 32 Tahun 2022 tentang Standar Biaya Umum Pemerintah Provinsi Jawa Timur Tahun 2023',
       'Penggolongan dan kodifikasi Barang Milik Daerah di Lingkungan Kabupaten Pemerintah Kabupaten Katingan',
       'Perubahan Kesembilan atas Peraturan Bupati Siak Nomor 177 Tahun 2019 tentang Penjabaran Anggaran Pendapatan dan Belanja Daerah Tahun Anggaran 2020'],
      dtype=object)

In [32]:
metadata_regulasi_df.to_csv('data/metadata_regulasi_penyiaran.csv', index=False)

## Riwayat Regulasi

In [33]:
riwayat_regulasi_df = pd.DataFrame(riwayat_regulasi)
riwayat_regulasi_df

data_id                                      judul_singkat  \
0      38966                               PP No. 52 Tahun 2005   
1      38952                               PP No. 50 Tahun 2005   
2      38955                               PP No. 51 Tahun 2005   
3      38618                               PP No. 11 Tahun 2005   
4      38618                               PP No. 11 Tahun 2005   
...      ...                                                ...   
2487  228836                 PERBUP Kab. Siak No. 29 Tahun 2020   
2488  175284  PERBUP Kab. Penajam Paser Utara No. 37 Tahun 2020   
2489  228833                 PERBUP Kab. Siak No. 26 Tahun 2020   
2490  228813                 PERBUP Kab. Siak No. 15 Tahun 2020   
2491  228834                 PERBUP Kab. Siak No. 27 Tahun 2020   

                                          judul_tentang relasi_peraturan  \
0     Penyelenggaraan Penyiaran Lembaga Penyiaran Be...  Diubah dengan :   
1     Penyelenggaraan Penyiaran Lembaga Penyiaran Sw...  Diubah dengan :   
2     Penyelenggaraan Penyiaran Lembaga Penyiaran Ko...  Diubah dengan :   
3     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...  Diubah dengan :   
4     Penyelenggaraan Penyiaran Lembaga Penyiaran Pu...  Diubah dengan :   
...                                                 ...              ...   
2487  Perubahan Kesepuluh atas Peraturan Bupati Siak...       Mengubah :   
2488  Perubahan Atas Peraturan Bupati Nomor 13 Tahun...       Mengubah :   
2489  Perubahan Kedelapan atas Peraturan Bupati Siak...       Mengubah :   
2490  Perubahan Ketiga atas Peraturan Bupati Siak No...       Mengubah :   
2491  Perubahan Kesembilan atas Peraturan Bupati Sia...       Mengubah :   

                                       regulasi_rujukan  \
0                                  PP No. 46 Tahun 2021   
1                                  PP No. 46 Tahun 2021   
2                                  PP No. 46 Tahun 2021   
3                                  PP No. 17 Tahun 2024   
4                                  PP No. 46 Tahun 2021   
...                                                 ...   
2487         PERATURAN BUPATI SIAK NOMOR 177 TAHUN 2019   
2488  PERBUP Kab. Penajam Paser Utara No. 13 Tahun 2019   
2489         PERATURAN BUPATI SIAK NOMOR 177 TAHUN 2019   
2490  Peraturan Bupati Siak No. 177 Tahun 2019 Tenta...   
2491         PERATURAN BUPATI SIAK NOMOR 177 TAHUN 2019   

                                       regulasi_tentang  
0                    Pos, Telekomunikasi, dan Penyiaran  
1                    Pos, Telekomunikasi, dan Penyiaran  
2                    Pos, Telekomunikasi, dan Penyiaran  
3     Perubahan atas Peraturan Pemerintah Nomor 11 T...  
4                    Pos, Telekomunikasi, dan Penyiaran  
...                                                 ...  
2487                                                     
2488       Rencana Kerja Pemerintah Daerah Tahun 2020\r  
2489                                                     
2490                                                     
2491                                                     

[2492 rows x 6 columns]

In [53]:
riwayat_regulasi_df.relasi_peraturan = riwayat_regulasi_df.relasi_peraturan.str.replace('dengan', '').str.replace(':', '').str.strip()
riwayat_regulasi_df.relasi_peraturan.unique()

array(['Diubah', 'Dicabut', 'Mencabut', 'Dicabut sebagian',
       'Mencabut sebagian', 'Mengubah', 'Mengubah sebagian',
       'Diubah sebagian', 'Menetapkan', 'Ditetapkan'], dtype=object)

In [54]:
riwayat_regulasi_df.to_csv('data/riwayat_regulasi_penyiaran.csv', index=False)

In [37]:
riwayat_uji_materi_df = pd.DataFrame(riwayat_uji_materi)
riwayat_uji_materi_df

data_id         judul_singkat  \
0    32772  UU No. 32 Tahun 2002   
1    28414  UU No. 10 Tahun 2008   
2    28414  UU No. 10 Tahun 2008   
3    28414  UU No. 10 Tahun 2008   
4    28414  UU No. 10 Tahun 2008   
5    29809  UU No. 11 Tahun 2006   
6    29809  UU No. 11 Tahun 2006   
7    26755  UU No. 42 Tahun 2008   
8    26755  UU No. 42 Tahun 2008   
9    26755  UU No. 42 Tahun 2008   
10   26755  UU No. 42 Tahun 2008   
11   26744   UU No. 7 Tahun 2017   
12   26744   UU No. 7 Tahun 2017   
13   26744   UU No. 7 Tahun 2017   
14   26744   UU No. 7 Tahun 2017   
15   26744   UU No. 7 Tahun 2017   
16   26744   UU No. 7 Tahun 2017   
17   26744   UU No. 7 Tahun 2017   
18   28116  UU No. 42 Tahun 2009   
19   34476   UU No. 3 Tahun 1997   
20   26438   UU No. 1 Tahun 2015   
21   27376   UU No. 8 Tahun 2015   
22   27376   UU No. 8 Tahun 2015   
23   27376   UU No. 8 Tahun 2015   
24   27376   UU No. 8 Tahun 2015   
25   27376   UU No. 8 Tahun 2015   
26   27376   UU No. 8 Tahun 2015   
27   27376   UU No. 8 Tahun 2015   
28   30504  UU No. 32 Tahun 2004   
29   30504  UU No. 32 Tahun 2004   
30   30504  UU No. 32 Tahun 2004   
31   30504  UU No. 32 Tahun 2004   
32   30504  UU No. 32 Tahun 2004   
33   26408  UU No. 10 Tahun 2016   
34   26408  UU No. 10 Tahun 2016   
35   26408  UU No. 10 Tahun 2016   
36   26408  UU No. 10 Tahun 2016   

                                        judul_tentang  \
0                                           Penyiaran   
1   Pemilihan Umum Anggota Dewan Perwakilan Rakyat...   
2   Pemilihan Umum Anggota Dewan Perwakilan Rakyat...   
3   Pemilihan Umum Anggota Dewan Perwakilan Rakyat...   
4   Pemilihan Umum Anggota Dewan Perwakilan Rakyat...   
5                                   Pemerintahan Aceh   
6                                   Pemerintahan Aceh   
7          Pemilihan Umum Presiden dan Wakil Presiden   
8          Pemilihan Umum Presiden dan Wakil Presiden   
9          Pemilihan Umum Presiden dan Wakil Presiden   
10         Pemilihan Umum Presiden dan Wakil Presiden   
11                                     Pemilihan Umum   
12                                     Pemilihan Umum   
13                                     Pemilihan Umum   
14                                     Pemilihan Umum   
15                                     Pemilihan Umum   
16                                     Pemilihan Umum   
17                                     Pemilihan Umum   
18  Perubahan Ketiga Atas Undang-Undang Nomor 8 Ta...   
19                                    Pengadilan Anak   
20  Penetapan Peraturan Pemerintah Pengganti Undan...   
21  Perubahan atas Undang-Undang Nomor 1 Tahun 201...   
22  Perubahan atas Undang-Undang Nomor 1 Tahun 201...   
23  Perubahan atas Undang-Undang Nomor 1 Tahun 201...   
24  Perubahan atas Undang-Undang Nomor 1 Tahun 201...   
25  Perubahan atas Undang-Undang Nomor 1 Tahun 201...   
26  Perubahan atas Undang-Undang Nomor 1 Tahun 201...   
27  Perubahan atas Undang-Undang Nomor 1 Tahun 201...   
28                                Pemerintahan Daerah   
29                                Pemerintahan Daerah   
30                                Pemerintahan Daerah   
31                                Pemerintahan Daerah   
32                                Pemerintahan Daerah   
33  Perubahan Kedua Atas Undang-Undang Nomor 1 Tah...   
34  Perubahan Kedua Atas Undang-Undang Nomor 1 Tah...   
35  Perubahan Kedua Atas Undang-Undang Nomor 1 Tah...   
36  Perubahan Kedua Atas Undang-Undang Nomor 1 Tah...   

                                        puu_mk_no  \
0                    PUTUSAN Nomor 005/PUU-I/2003   
1                 PUTUSAN Nomor 22-24/PUU-VI/2008   
2                    PUTUSAN Nomor 12/PUU-VI/2008   
3                  PUTUSAN Nomor 27/PUU-VIII/2010   
4                  PUTUSAN Nomor 12/PUU-VIII/2010   
5                   PUTUSAN Nomor 51/PUU-XIV/2016   
6                  PUTUSAN Nomor 35/PUU-VIII/2010   
7                   PUTUSAN Nomor 98/PUU-VI

In [38]:
riwayat_uji_materi_df.to_csv('data/riwayat_uji_materi_penyiaran.csv', index=False)

Trial and Error

In [33]:
# abstrak.find('div', attrs={'class':'fs-5'}).find_all('p')[0].text.strip() # judul singkat
# abstrak.find('div', attrs={'class':'fs-5'}).find_all('p')[1].text.strip() # tahun
# abstrak.find('div', attrs={'class':'fs-5'}).find_all('p')[2].text.strip() # sumber
# abstrak.find('div', attrs={'class':'fs-5'}).find_all('p')[3].text.strip() # judul lengkap

# abstrak.find_all('tr')[0].find_all('li')[0].text.strip() # pertimbangan
# abstrak.find_all('tr')[0].find_all('li')[1].text.strip() # konsideran
# abstrak.find_all('tr')[0].find_all('li')[2].text.replace('\r\n','').strip() # materi pokok

# print('\n'.join([x.text for x in abstrak.find_all('tr')[1].find_all('li')])) # catatan

'Undang-undang (UU) tentang Penyiaran'